src/editors/containers/TextEditor/components/SourceCodeModal/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  FormattedMessage,
+  injectIntl,
+  intlShape,
+} from '@edx/frontend-platform/i18n';
+
+import { Button } from '@edx/paragon';
+import messages from './messages';
+import hooks from './hooks';
+import BaseModal from '../BaseModal';
+
+import CodeEditor from '../RawEditor/CodeEditor';
+
+export const SourceCodeModal = ({
+  isOpen,
+  close,
+  editorRef,
+  // injected
+  intl,
+}) => {
+  const { saveBtnProps, value, ref } = hooks.preprareSourceCodeModal({ editorRef, close });
+  return (
+    <BaseModal
```

Comment:
I think we would want this Modal to be bigger than the image upload one. A size prop could be added to 'BaseModal.js` so it could set the `ModalDialog` paragon component with the desired size.